librz/core/linux_heap_glibc.c

```diff
@@ -391,19 +398,26 @@ static void GH(print_arena_stats)(RzCore *core, GHT m_arena, MallocState *main_a
 	PRINT_GA("}\n\n");
 }
 
-static bool GH(rz_resolve_main_arena)(RzCore *core, GHT *m_arena) {
+/**
+ * @brief Store the base address of main arena at m_arena
+ * @param core RzCore pointer
+ * @param m_arena Store the location of main arena at this integer pointer
+ * @return True if a main arena was found else False
+ */
+RZ_API bool GH(rz_resolve_main_arena)(RzCore *core, GHT *m_arena) {
 	rz_return_val_if_fail(core && core->dbg && core->dbg->maps, false);
 
-	if (core->dbg->main_arena_resolved) {
-		return true;
-	}
+	//	if (core->dbg->main_arena_resolved) {
```

Comment:
Why this was commented? Is this still necessary? If not, lets remove the commented code. cc @thestr4ng3r @ret2libc

Reply:
During the refactoring, I tried to get the code with this statement uncommented but I was facing some bugs in which `m_arena` was not be initialized properly. Due to the nature of the refactor `GHT m_arena` is declared multiple times so we cannot use this hack to save performance anymore.